next/tp6/discretization.py

```diff
@@ -1,85 +1,124 @@
-'''
-Helper class to systematically convert a R^n vector of float into a 
+"""
+Helper class to systematically convert a R^n vector of float into a
 N^n vector of integers, and to a single integer.
 The main methods are flat2continuous and continuous2flat.
 Build it from the size of the real vector space, the bounds in each directions and
 the number of discretization steps.
 
 See also numpy.digitalize.
-'''
+"""
 
-import numpy as np
 from functools import reduce
 
+import numpy as np
+
+
 class VectorDiscretization:
-    '''
+    """
     Convertion between continuous R^N bounded vectors and their discretization
     as N^N bounded integer vectors and N*N integer.
-    '''
-    def __init__(self,nv,vmax=1.,vmin=None,nsteps=10,modulo=None,moduloIdx=[]):
+    """
+
+    def __init__(self, nv, vmax=1.0, vmin=None, nsteps=10, modulo=None, moduloIdx=[]):
         self.nv = nv  # Dimension of the vector space
-        self.vmax = vmax if isinstance(vmax,np.ndarray) else np.array([vmax,]*nv)
-        self.vmin = vmin if isinstance(vmin,np.ndarray) else -self.vmax if vmin is None else np.array([vmin,]*nv)
-        self.nsteps = nsteps if isinstance(nsteps,np.ndarray) else np.array([nsteps,]*nv)
-        self.nd = reduce(lambda i,j:i*j,self.nsteps)  # prod_{s in nsteps} s = s1*s2*...*s_nv
+        self.vmax = (
+            vmax
+            if isinstance(vmax, np.ndarray)
+            else np.array(
+                [
+                    vmax,
+                ]
+                * nv
+            )
+        )
+        self.vmin = (
+            vmin
+            if isinstance(vmin, np.ndarray)
+            else -self.vmax
+            if vmin is None
+            else np.array(
+                [
+                    vmin,
+                ]
+                * nv
+            )
```

Comment:
```suggestion
            else np.array([vmin]  * nv )
```
if linters are OK with this. same below.